OpenTrons/opentrons-api
# feat(app, labware-library, app-shell): Labware creator in app prototype

# Overview

While working on LC the past few weeks I realized that putting it in the app is actually a fairly easy task. This is the _InT3rNaL pRoTOtyP3_ 🤫 and it is behind the feature flag ❤️ 

https://github.com/Opentrons/opentrons/assets/66035149/04a533a0-e7d9-483b-8860-969134094b52


# Test Plan

- build ur dependencies!
- Turn on the Enable App Labware Creator feature flag
- go to the Labware tab
- click on the Open Labware Creator button
- it should launch a wizard with the LC pages on each page and skipping the pages if the specs you've entered don't work
- there should be some error handling where the banner pops up if your field is missing or incorrect and the next button should be disabled if that is the case
- once you create the definition, it should auto-populate in the Custom Labware section of the Labware library
- click on that labware and open the slideout and inspect the slideout specs
- then launch the wizard again and click on "upload a file that you've created" and upload a labware file (it does not direct to the folder that your labware was added to -- need to think about how to approach that more) but after you upload it should populate in the wizard
- test labware-creator in the dev environment to make sure it has retained the designs and functionality

# Changelog

- make feature flag
- create modal branching logic in LC 
- make LL a module that app package.json imports
- edit the configs so LL is treated like a module
- add the button to `Labware` in the app
- add some new redux selectors in app-shell and app so they it takes in a string of the json file instead of importing a file
- fix cypress tests
- fix unit tests

# Review requests



# Risk assessment

low, it is behind a ff
